homeassistant/components/vizio/media_player.py

```diff
@@ -32,35 +45,56 @@
 MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
 
-SUPPORTED_COMMANDS = SUPPORT_TURN_ON | SUPPORT_TURN_OFF \
-                     | SUPPORT_SELECT_SOURCE \
-                     | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK \
-                     | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP \
-                     | SUPPORT_VOLUME_SET
+COMMON_SUPPORTED_COMMANDS = (
+    SUPPORT_SELECT_SOURCE |
+    SUPPORT_TURN_ON |
+    SUPPORT_TURN_OFF |
+    SUPPORT_VOLUME_MUTE |
+    SUPPORT_VOLUME_SET |
+    SUPPORT_VOLUME_STEP
+)
+
+SUPPORTED_COMMANDS = {
+    'soundbar': COMMON_SUPPORTED_COMMANDS,
+    'tv': (
+        COMMON_SUPPORTED_COMMANDS |
+        SUPPORT_NEXT_TRACK |
+        SUPPORT_PREVIOUS_TRACK
+    )
+}
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_HOST): cv.string,
-    vol.Required(CONF_ACCESS_TOKEN): cv.string,
+    vol.Optional(CONF_ACCESS_TOKEN): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_SUPPRESS_WARNING, default=False): cv.boolean,
+    vol.Optional(CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS):
+        vol.All(cv.string, vol.Lower, vol.In(['tv', 'soundbar'])),
     vol.Optional(CONF_VOLUME_STEP, default=DEFAULT_VOLUME_STEP):
         vol.All(vol.Coerce(int), vol.Range(min=1, max=10)),
 })
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the VizioTV media player platform."""
+    """Set up the Vizio media player platform."""
     host = config.get(CONF_HOST)
     token = config.get(CONF_ACCESS_TOKEN)
     name = config.get(CONF_NAME)
     volume_step = config.get(CONF_VOLUME_STEP)
-
-    device = VizioDevice(host, token, name, volume_step)
+    device_type = config.get(CONF_DEVICE_CLASS)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.